### PR TITLE
Matt Simon

### PR DESCRIPTION
**Pull request by msimon42**

### CODE REVIEW

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @patrons << patron
+    exhibits = self.recommend_exhibits(patron)
+    exhibits.sort_by {|exhibit| exhibit.cost}
+    exhibits.each do |exhibit|
+      if exhibit.cost <= patron.spending_money
+        @patrons_of_exhibits[exhibit] << patron
+        patron.spending_money -= exhibit.cost
+        @revenue += exhibit.cost
+      end
+    end
```
Comment (dionew1) [inline lib/museum.rb]: This method looks like it is handling a lot of logic and could be refactored for easier readability.

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  def patrons_by_exhibit_interest
+    interest_list = Hash.new
+    @exhibits.each do |exhibit|
+      interest_list[exhibit] = []
+      @patrons.each do |patron|
+        if self.recommend_exhibits(patron).include?(exhibit)
+          interest_list[exhibit] << patron
+        end
+      end
+    end
```
Comment (dionew1) [inline lib/museum.rb]: Avoid nesting `each`s

File: test/museum_test.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
+require 'minitest'
+require 'minitest/autorun'
+require_relative '../lib/museum'
+require_relative '../lib/exhibit'
+require_relative '../lib/patron'
+
+class MuseumTest < Minitest::Test
```
Comment (dionew1) [inline test/museum_test.rb]: Great job on writing your tests!

File: lib/exhibit.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Exhibit
+  attr_reader :name, :cost
+
```
Comment (dionew1) [inline lib/exhibit.rb]: Nice job!

File: lib/museum.rb (hunk 2 above)
Comment (msimon42) [reply lib/museum.rb]: What would be a good alternative?

**Pull request closed without merging**